vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -173,39 +173,38 @@ void Promotion::OnGetAllPromotions(
     return;
   }
 
-  if (list.size() > 0) {
-    for (auto & item : list) {
-      // if the server return expiration for ads we need to set it to 0
-      if (item->type == ledger::PromotionType::ADS) {
-        item->expires_at = 0;
-      }
+  for (auto & item : list) {
+    auto it = promotions.find(item->id);
+    if (it != promotions.end() &&
+        it->second->status != ledger::PromotionStatus::ACTIVE) {
+      continue;
+    }
 
-      if (item->legacy_claimed) {
-        item->status = ledger::PromotionStatus::CLAIMED;
```

Comment:
this is where the bug was as we didn't check if we already started claim process for the promotion.